src/manager/agent.manager.service.ts

```diff
@@ -120,9 +120,13 @@ export class AgentManagerService {
      * TODO handle case were agent not there
      */
     public async spinDownAgent(agentId: string) {
-        Logger.log('Spinning down agent', agentId);
-        await this.cache.del(agentId);
-        await this.manager.stopAgent(agentId);
+        try {
+            Logger.log('Spinning down agent', agentId);
+            await this.cache.del(agentId);
+            await this.manager.stopAgent(agentId);
```

Comment:
should we put each statement in its own "try/catch".  we want to clear the cache.  if that fails, we still want to stop the agent.

Reply:
I think it's fine to keep it like this -  if we wanted to be extra careful we'd wrap each call in a try-catch but that seems overkill at this point